### PR TITLE
fix: tolerate workflow that needs a missing job (#1595)

### LINKED ISSUE #1595: Unable to build dependency graph
### Bug report info

```plain text
act version:            0.2.41
GOOS:                   darwin
GOARCH:                 amd64
NumCPU:                 12
Docker host:            DOCKER_HOST environment variable is unset/empty.
Sockets found:
	/var/run/docker.sock
	/Users/srepollock/.docker/run/docker.sock
Config files:           
Build info:
	Go version:            go1.19.5
	Module path:           command-line-arguments
	Main version:          
	Main path:             
	Main checksum:         
	Build settings:
		-compiler:            gc
		-ldflags:             -X main.version=0.2.41
		CGO_ENABLED:          1
		CGO_CFLAGS:           
		CGO_CPPFLAGS:         
		CGO_CXXFLAGS:         
		CGO_LDFLAGS:          
		GOARCH:               amd64
		GOOS:                 darwin
		GOAMD64:              v1
Docker Engine:
	Engine version:        20.10.22
	Engine runtime:        runc
	Cgroup version:        2
	Cgroup driver:         cgroupfs
	Storage driver:        overlay2
	Registry URI:          https://index.docker.io/v1/
	OS:                    Docker Desktop
	OS type:               linux
	OS version:            
	OS arch:               x86_64
	OS kernel:             5.15.49-linuxkit
	OS CPU:                4
	OS memory:             3933 MB
	Security options:
		name=seccomp,profile=default
		name=cgroupns
```

### Command used with act

```sh
act --verbose
```

### Describe issue

I expected the module to run any of my programs

### Link to GitHub repository

_No response_

### Workflow content

```yml
n/a
```

### Relevant log output

```sh
DEBU[0000] Loading environment from /redacted/.env
DEBU[0000] Loading action inputs from /redacted/.input
DEBU[0000] Loading secrets from /redacted/.secrets
DEBU[0000] Loading workflows from '/redacted/.github/workflows'
DEBU[0000] Loading workflows recursively
DEBU[0000] Found workflow 'build.yml' in '/redacted/.github/workflows/build.yml'
DEBU[0000] Found workflow 'release.yml' in '/redacted/.github/workflows/release.yml'
DEBU[0000] Found workflow 'test.yml' in '/redacted/.github/workflows/test.yml'
DEBU[0000] Reading workflow '/redacted/.github/workflows/build.yml'
DEBU[0000] Reading workflow '/redacted/.github/workflows/release.yml'
DEBU[0000] Reading workflow '/redacted/.github/workflows/test.yml'
DEBU[0000] Preparing plan with all jobs
FATA[0000] Unable to build dependency graph!
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jsoref**
Try to fix #1595...

## Commit Message

{{title}} (#1619)

Change planner functions to return errors

This enables createStages to return `unable to build dependency graph`

Fix PlanEvent to properly report errors relating to events/workflows

### CODE REVIEW

**Comment by jsoref** [comment]
With this, I'm able to run `act -l` for the vector repository and get something reasonable

<details><summary>output</summary>

```
jsoref@jsoref-mbp vector % ../nektos/act/dist/local/act -l
WARN  ⚠ You are using Apple M1 chip and you have not specified container architecture, you might encounter issues while running act. If so, try running it with '--container-architecture linux/amd64'. ⚠
WARN[0000] unable to build dependency graph for Test Harness (test-harness.yml)
Stage  Job ID                                         Job name                                                                                                                Workflow name                  Workflow file            Events
0      release-build-normal                           Release Build (normal)                                                                                                  Baseline Timings               baseline.yml             workflow_dispatch,schedule
0      debug-build                                    Debug Build                                                                                                             Baseline Timings               baseline.yml             workflow_dispatch,schedule
0      debug-rebuild                                  Debug Rebuild                                                                                                           Baseline Timings               baseline.yml             schedule,workflow_dispatch
0      check                                          Cargo Check                                                                                                             Baseline Timings               baseline.yml             workflow_dispatch,schedule
0      release-build-optimized                        Release Build (optimized)                                                                                               Baseline Timings               baseline.yml             workflow_dispatch,schedule
0      publish-new-environment                        publish-new-environment                                                                                                 Environment Suite              environment.yml          pull_request,push,workflow_dispatch
0      add-to-project                                 Add issue to Gardener project board                                                                                     gardener_open_issue.yml        gardener_open_issue.yml  issues
0      add-dependabot-to-project                      Add dependabot PR to Gardener project board                                                                             gardener_open_pr.yml           gardener_open_pr.yml     pull_request_target
0      add-contributor-to-project                     Add contributor PR to Gardener project board                                                                            gardener_open_pr.yml           gardener_open_pr.yml     pull_request_target
0      sync-install                                   sync-install                                                                                                            Update install.sh Suite        install-sh.yml           push,workflow_dispatch
0      test-integration                               Integration - Linux, ${{ matrix.test }}                                                                                 Integration Test Suite         integration-test.yml     workflow_dispatch,push,pull_request
0      build-x86_64-unknown-linux-gnu                 Build - x86_64-unknown-linux-gnu                                                                                        K8S E2E Suite                  k8s_e2e.yml              workflow_dispatch,push,pull_request
0      compute-k8s-test-plan                          Compute K8s test plan                                                                                                   K8S E2E Suite                  k8s_e2e.yml              workflow_dispatch,push,pull_request
0      label                                          label                                                                                                                   Pull Request Labeler           labeler.yml              pull_request_target
0      Nightly                                        Nightly                                                                                                                 Nightly                        nightly.yml              workflow_dispatch,schedule
0      generate-publish-metadata                      Generate Publish-related Metadata                                                                                       Publish                        publish.yml              workflow_call
0      cancel-previous                                cancel-previous                                                                                                         Regression Detector            regression.yml           pull_request
0      compute-metadata                               Compute metadata for regression experiments                                                                             Regression Detector            regression.yml           pull_request
0      compute-metadata                               Compute metadata for regression experiments                                                                             Regression Detector (trusted)  regression_trusted.yml   workflow_run
0      Release                                        Release                                                                                                                 Release Suite                  release.yml              push
0      spelling                                       Check Spelling                                                                                                          Check Spelling                 spelling.yml             push,pull_request_target
0      changes                                        changes                                                                                                                 Test Suite                     test.yml                 pull_request,push
1      test-install                                   test-install                                                                                                            Update install.sh Suite        install-sh.yml           push,workflow_dispatch
1      test-integration-check                         test-integration-check                                                                                                  Integration Test Suite         integration-test.yml     push,pull_request,workflow_dispatch
1      test-e2e-kubernetes                            K8s ${{ matrix.kubernetes_version.version }} / ${{ matrix.container_runtime }} (${{ matrix.kubernetes_version.role }})  K8S E2E Suite                  k8s_e2e.yml              push,pull_request,workflow_dispatch
1      build-x86_64-pc-windows-msvc-packages          Build Vector for x86_64-pc-windows-msvc (.zip)                                                                          Publish                        publish.yml              workflow_call
1      build-armv7-unknown-linux-gnueabihf-packages   Build Vector for armv7-unknown-linux-gnueabihf (.tar.gz)                                                                Publish                        publish.yml              workflow_call
1      build-aarch64-unknown-linux-gnu-packages       Build Vector for aarch64-unknown-linux-gnu (.tar.gz)                                                                    Publish                        publish.yml              workflow_call
1      build-x86_64-unknown-linux-musl-packages       Build Vector for x86_64-unknown-linux-musl (.tar.gz)                                                                    Publish                        publish.yml              workflow_call
1      build-x86_64-apple-darwin-packages             Build Vector for x86_64-apple-darwin (.tar.gz)                                                                          Publish                        publish.yml              workflow_call
1      build-aarch64-unknown-linux-musl-packages      Build Vector for aarch64-unknown-linux-musl (.tar.gz)                                                                   Publish                        publish.yml              workflow_call
1      build-armv7-unknown-linux-musleabihf-packages  Build Vector for armv7-unknown-linux-musleabihf (.tar.gz)                                                               Publish                        publish.yml              workflow_call
1      build-x86_64-unknown-linux-gnu-packages        Build Vector for x86_64-unknown-linux-gnu (.tar.gz, DEB, RPM)                                                           Publish                        publish.yml              workflow_call
1      build-baseline                                 Build baseline Vector container                                                                                         Regression Detector            regression.yml           pull_request
1      build-comparison                               Build comparison Vector container                                                                                       Regression Detector            regression.yml           pull_request
1      transmit-metadata                              Transmit metadata to trusted workflow                                                                                   Regression Detector            regression.yml           pull_request
1      confirm-valid-credentials                      Confirm AWS credentials are minimally valid                                                                             Regression Detector (trusted)  regression_trusted.yml   workflow_run
1      test-linux                                     Unit - x86_64-unknown-linux-gnu                                                                                         Test Suite                     test.yml                 pull_request,push
1      test-misc                                      Miscellaneous - Linux                                                                                                   Test Suite                     test.yml                 pull_request,push
1      test-windows                                   Unit - Windows                                                                                                          Test Suite                     test.yml                 pull_request,push
1      checks                                         Checks                                                                                                                  Test Suite                     test.yml                 pull_request,push
1      check-component-features                       Component Features - Linux                                                                                              Test Suite                     test.yml                 pull_request,push
1      test-cli                                       CLI - Linux                                                                                                             Test Suite                     test.yml                 pull_request,push
1      cross-linux                                    Cross - ${{ matrix.target }}                                                                                            Test Suite                     test.yml                 push,pull_request
1      test-mac                                       Unit - Mac                                                                                                              Test Suite                     test.yml                 push,pull_request
1      check-msrv                                     Check minimum supported Rust version                                                                                    Test Suite                     test.yml                 pull_request,push
1      test-vrl-wasm                                  VRL WASM - Linux                                                                                                        Test Suite                     test.yml                 pull_request,push
1      test-vrl                                       VRL - Linux                                                                                                             Test Suite                     test.yml                 pull_request,push
2      install-shell-failure                          install-shell-failure                                                                                                   Update install.sh Suite        install-sh.yml           push,workflow_dispatch
2      master-failure                                 master-failure                                                                                                          Integration Test Suite         integration-test.yml     workflow_dispatch,push,pull_request
2      master-failure                                 master-failure                                                                                                          K8S E2E Suite                  k8s_e2e.yml              workflow_dispatch,push,pull_request
2      deb-verify                                     Verify DEB Packages                                                                                                     Publish                        publish.yml              workflow_call
2      macos-verify                                   Verify macOS Package                                                                                                    Publish                        publish.yml              workflow_call
2      rpm-verify                                     Verify RPM Packages                                                                                                     Publish                        publish.yml              workflow_call
2      upload-baseline-image-to-ecr                   Upload images to ECR                                                                                                    Regression Detector (trusted)  regression_trusted.yml   workflow_run
2      upload-comparison-image-to-ecr                 Upload images to ECR                                                                                                    Regression Detector (trusted)  regression_trusted.yml   workflow_run
2      master-failure                                 master-failure                                                                                                          Test Suite                     test.yml                 pull_request,push
2      cross-linux-check                              Cross - Linux                                                                                                           Test Suite                     test.yml                 pull_request,push
3      publish-github                                 Publish to GitHub                                                                                                       Publish                        publish.yml              workflow_call
3      publish-cloudsmith                             Publish to Cloudsmith                                                                                                   Publish                        publish.yml              workflow_call
3      publish-docker                                 Publish to Docker                                                                                                       Publish                        publish.yml              workflow_call
3      publish-s3                                     Publish to S3                                                                                                           Publish                        publish.yml              workflow_call
3      submit-job                                     Submit regression job                                                                                                   Regression Detector (trusted)  regression_trusted.yml   workflow_run
4      publish-homebrew                               Publish to Homebrew                                                                                                     Publish                        publish.yml              workflow_call
4      publish-failure                                Send Publish Failure Notification                                                                                       Publish                        publish.yml              workflow_call
4      detect-regression                              Determine regression status                                                                                             Regression Detector (trusted)  regression_trusted.yml   workflow_run
4      analyze-experiment                             Download regression analysis & upload report                                                                            Regression Detector (trusted)  regression_trusted.yml   workflow_run

Detected multiple jobs with the same job name, use `-W` to specify the path to the specific workflow.
```

</details>

File: pkg/model/planner.go [hunk 1]
```diff
@@ -169,47 +169,65 @@ type workflowPlanner struct {
 }
 
 // PlanEvent builds a new list of runs to execute in parallel for an event name
-func (wp *workflowPlanner) PlanEvent(eventName string) *Plan {
+func (wp *workflowPlanner) PlanEvent(eventName string) (*Plan, error) {
 	plan := new(Plan)
 	if len(wp.workflows) == 0 {
-		log.Debugf("no events found for workflow: %s", eventName)
+		log.Debugf("no workflows found for workflow: %s", eventName)
```
Comment (KnisterPeter) [inline pkg/model/planner.go:175]: This should probably be

```suggestion
		log.Debugf("no workflows found for event: %s", eventName)
```

**Review by KnisterPeter — CHANGES_REQUESTED** [review]
Looks fine, just the debug output seems to be wrong

**Review by KnisterPeter — COMMENTED** [review]
I'm fine with this now, but @ChristopherHX has a valid point.
If that's fixed I'll approve

**Comment by jsoref** [comment]
I'm happy to apply most of that. This seems unnecessary:

```
-		log.Debug("no workflows found by planner")
+		log.Debugf("no workflows found by planner")
```

**Comment by jsoref** [comment]
(I'm just trying to convince my computer to run some tests in VSCode before I amend.)

**Review by KnisterPeter — APPROVED** [review]

**Pull request merged**